### ISSUE TITLE
onbeforesave saving to model

### ISSUE THREAD

**Issue by QuijoteShin** [vitalets/angular-xeditable#161]
Hi,

I'm facing a posible bug.

I'm trying to emulate $pristine 
Só i'm at onshow storing the  unmodified $scope.user  to a local variable userBack

Then onbeforesave I comparing each  $scope.user item to userBack.
 The problem is that If I modify $scope.user , userBack are become updated even when is supposed to be updated only at onshow

Debugging:  if I return 'string' at onbeforesave , userBack didn't change either.

Basically I think is saving to model at onbeforesave,

Actually if I place a console.log($scope.user) at the beggining and at the success event of your example ( http://jsfiddle.net/NfPcH/81/  ) model $scope.user and the userBack at the beggining  are already modified, 

You can check this here: http://jsfiddle.net/NfPcH/2841/
